rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/IDEServicesThread.java

```diff
@@ -42,14 +42,16 @@
     private final IBaseLanguageClient ideClient;
     private final ServerSocket serverSocket;
     private final IBaseTextDocumentService docService;
+    private final BaseWorkspaceService workspaceService;
     private static final Logger logger = LogManager.getLogger(IDEServicesThread.class);
 
-    public IDEServicesThread(IBaseLanguageClient client, IBaseTextDocumentService docService, ServerSocket socket) {
+    public IDEServicesThread(IBaseLanguageClient client, IBaseTextDocumentService docService, BaseWorkspaceService workspaceService, ServerSocket socket) {
```

Comment:
Happy you picked that solution instead of extending IBaseTextDocumentService. cool!

Reply:
thanks, it was the way to go, the IBaseTextDocumentService is already too big.